liota/dccs/iotcc.py

```diff
@@ -536,8 +563,17 @@ def store_reg_entity_attributes(self, entity_type, entity_name, reg_entity_id,
                 new_prop_dict = tmp_dict
         else:
             tmp_dict = self.read_entity_file(reg_entity_id)
-            if ((tmp_dict["entity type"] == entity_type) and (tmp_dict["name"] == entity_name)
-                and (tmp_dict["device type"] == dev_type)):
+            # check Entity_Timestamp of entity_file: if < boottime, get properties from cloud
+            if (self.enable_reboot_getprop == "True") and ('Entity_Timestamp' in tmp_dict):
+                last_dtime = datetime.datetime.strptime(tmp_dict["Entity_Timestamp"], "%Y-%m-%dT%H:%M:%S")
+                if (last_dtime <= self.boottime):
+                    list_prop = self.get_properties(reg_entity_id)
+                    # merge property info from get_properties() into our local entity record
+                    tmp_dict = self.merge_prop_dict_list(tmp_dict, list_prop)
+            if ((('entity type' in tmp_dict) and (tmp_dict["entity type"] == entity_type)) and
```

Comment:
Can you please explain this if condition? It is so hard to understand. We are checking for None for some, '' for some, there are a lot of and and or statements...

Reply:
In fact, it is quite straightforward. It checked 'entity type', 'entity_name', 'device type' of this fun's inputs with those inside local record, and see whether they are for the same entity or not.
(('entity type' in tmp_dict) and (tmp_dict["entity type"] == entity_type)) is for 'entity type'.
front half is guarantee that 'entity type' is inside the local record.

For the 'entity name', it is similar.
For 'device type', it is a little different since we have "EdgeSystem" and "Connected Device". For 'EdgeSystem', our local record will be '' and the input dev_type should be None.

In fact, I have the comment under the if () # the same entity.